src/mon/OSDMonitor.cc

```diff
@@ -7784,22 +7784,37 @@ int OSDMonitor::get_crush_rule(const string &rule_name,
   return 0;
 }
 
-int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, ostream *ss)
+int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, int crush_rule, ostream *ss)
 {
   auto max_pgs_per_osd = g_conf().get_val<uint64_t>("mon_max_pg_per_osd");
-  auto num_osds = std::max(osdmap.get_num_in_osds(), 3u);   // assume min cluster size 3
-  auto max_pgs = max_pgs_per_osd * num_osds;
   uint64_t projected = 0;
-  if (pool < 0) {
```

Comment:
also, you need to take the not-yet-exist pool into consideration.

Reply:
Sorry, I didn't consider the pool >= 0 case. In my opinion, when pool < 0 the pg_num * size of the new pool should be added directly, and when pool >= 0 the current number of pg_num*size of the pool should be subtracted.